oidc/verification.go

```diff
@@ -142,29 +150,49 @@ func VerifyClientClaims(jwt jose.JWT, issuer string) (string, error) {
 	return sub, nil
 }
 
-type JWTVerifier struct {
-	issuer   string
-	clientID string
-	syncFunc func() error
-	keysFunc func() []key.PublicKey
-	clock    clockwork.Clock
+type JWTVerifier interface {
+	Verify(jose.JWT) error
 }
 
+type jwtVerifier struct {
+	issuer    string
+	clientID  string
+	syncFunc  func() error
+	keysFunc  func() []key.PublicKey
+	clock     clockwork.Clock
+	verifyAud bool
+}
+
+// NewJWTVerifier creates a verifier that validates an ID Token in accordance with
+// http://openid.net/specs/openid-connect-basic-1_0.html#IDTokenValidation
 func NewJWTVerifier(issuer, clientID string, syncFunc func() error, keysFunc func() []key.PublicKey) JWTVerifier {
-	return JWTVerifier{
-		issuer:   issuer,
-		clientID: clientID,
-		syncFunc: syncFunc,
-		keysFunc: keysFunc,
-		clock:    clockwork.NewRealClock(),
+	return &jwtVerifier{
+		issuer:    issuer,
+		clientID:  clientID,
+		syncFunc:  syncFunc,
+		keysFunc:  keysFunc,
+		clock:     clockwork.NewRealClock(),
+		verifyAud: true,
+	}
+}
+
+// NewJWTVerifierRelaxed creates a verifier that follows OpenID Connect ID Token validation
+// procedure except for not validating the 'aud' claim.
+func NewJWTVerifierRelaxed(issuer string, syncFunc func() error, keysFunc func() []key.PublicKey) JWTVerifier {
```

Comment:
> To be fair, I don't like it either. I had thought about the name for a while, and this one is the best I came up with. Maybe you have some suggestions?

I'll think about it, but the biggest part of any change like this is proposing the exact API changes. The naming, creating an API that might support additional validation tuning, etc. are things we should think about particularly since once it's in it's set in stone.